### PR TITLE
Fix incorrect display in reminder list label

### PR DESCRIPTION
**Pull request by Lainow**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12152

Corrected display of dates in reminder lists. 
imageimage
For example, previously, if the item creation date was 2023-10-28 23:59:59.000 and the current date was 2023-10-30 07:00:00.000, the label displayed was not 2 days ago but yesterday.

This pull request changes the way days are counted, retaining only the date without the time. It also adds week and month management.
imageimage

### CODE REVIEW

**Review by Rom1-B — APPROVED** [review]
Seems good

**Comment by trasher** [comment]
Honestly... This is really too tricky for me (and it's javascript, not my cup of tea)

**Comment by trasher** [comment]
(also: I guess this kind of stuff is a good candidate for a test)

File: js/common.js [hunk 1]
```diff
@@ -1166,9 +1173,13 @@ function relativeDate(str) {
                     : (tmp = Math.round(h)) === 1 ? __('an hour ago')
                         : d < 1.01 ? '%s hours ago'.replace('%s', tmp)
```
Comment (cedric-anne) [inline js/common.js:1174]: Looks like all `%s xxx ago` strings are not translated here.

All occurences should be fixed like this, I guess:
```suggestion
                        : d < 1.01 ? __('%s hours ago').replace('%s', tmp)
```

File: js/common.js [hunk 2]
```diff
@@ -1166,9 +1173,13 @@ function relativeDate(str) {
...
                             : (tmp = Math.round(d)) === 1 ? __('yesterday')
-                                : y < 1.01 ? '%s days ago'.replace('%s', tmp)
-                                    : (tmp = Math.round(y)) === 1 ? __('a year ago')
-                                        : '%s years ago'.replace('%s', tmp);
+                                : w < 1.01 ? '%s days ago'.replace('%s', tmp)
+                                    : (tmp = Math.floor(w)) === 1 ? __('a week ago')
+                                        : mo < 1.01 ? '%s weeks ago'.replace('%s', tmp)
+                                            : (tmp = Math.floor(mo)) === 1 ? __('a month ago')
+                                                : y < 1 ? '%s month ago'.replace('%s', tmp)
```
Comment (cedric-anne) [inline js/common.js:1180]:
```suggestion
                                                : y < 1 ? '%s months ago'.replace('%s', tmp)
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Pull request merged**